core/ledger/kvledger/txmgmt/statedb/statecouchdb/commit_handling_test.go

```diff
@@ -80,3 +81,67 @@ func TestGetRevision(t *testing.T) {
 	assert.Equal(t, "", revisionsMap["key-in-db"])
 
 }
+
+func TestBuildCommittersForNs(t *testing.T) {
+	env := testEnv
+	env.init(t, &statedb.Cache{})
+	defer env.cleanup()
+
+	versionedDB, err := testEnv.DBProvider.GetDBHandle("test-build-committers-for-ns")
+	assert.NoError(t, err)
+	db := versionedDB.(*VersionedDB)
+
+	nsUpdates := map[string]*statedb.VersionedValue{
+		"bad-key": {},
+	}
+
+	_, err = db.buildCommittersForNs("ns", nsUpdates)
+	assert.EqualError(t, err, "invalid namespace update version [<nil>]")
+
+	nsUpdates = make(map[string]*statedb.VersionedValue)
+	// populate updates with 1001 entries, default match batch update size is 1000, this requires 2 batches.
+	dummyHeight := version.NewHeight(1, 1)
+	for i := 0; i <= 1000; i++ {
+		nsUpdates[strconv.Itoa(i)] = &statedb.VersionedValue{
+			Value:    nil,
+			Metadata: nil,
+			Version:  dummyHeight,
+		}
+	}
+
+	committers, err := db.buildCommittersForNs("ns", nsUpdates)
+	assert.NoError(t, err)
+	assert.Equal(t, 2, len(committers))
+	assert.Equal(t, "ns", committers[0].namespace)
+
+}
+
+func TestBuildCommitters(t *testing.T) {
+	env := testEnv
+	env.init(t, &statedb.Cache{})
+	defer env.cleanup()
+
+	versionedDB, err := testEnv.DBProvider.GetDBHandle("test-build-committers")
+	assert.NoError(t, err)
+	db := versionedDB.(*VersionedDB)
+
+	batch := statedb.NewUpdateBatch()
+	batch.Put("ns-1", "key1", []byte("value1"), version.NewHeight(1, 1))
+	batch.Put("ns-2", "key1", []byte("value2"), version.NewHeight(1, 1))
```

Comment:
It will be good to add maxBatch+1 number of keys to one of the namespaces and eventually check for three committers.